tree/ntuple/v7/test/ntuple_basics.cxx

```diff
@@ -639,3 +639,52 @@ TEST(RNTuple, BareEntry)
    ntuple->LoadEntry(1);
    EXPECT_EQ(2.0, *ntuple->GetModel()->GetDefaultEntry()->Get<float>("pt"));
 }
+
+namespace ROOT::Experimental::Internal {
+struct RFieldCallbackInjector {
+   template <typename FieldT>
+   static void Inject(FieldT &field, ROOT::Experimental::Detail::RFieldBase::ReadCallback_t func)
+   {
+      field.AddReadCallback(func);
+   }
+};
+} // namespace ROOT::Experimental::Internal
+namespace {
+static unsigned gNCallReadCallback = 0;
+}
+
+TEST(RNTuple, ReadCallback)
+{
+   using RFieldCallbackInjector = ROOT::Experimental::Internal::RFieldCallbackInjector;
+
+   FileRaii fileGuard("test_ntuple_readcb.ntuple");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldI32 = model->MakeField<std::int32_t>("i32", 0);
+      auto fieldKlass = model->MakeField<CustomStruct>("klass");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "f", fileGuard.GetPath());
+      ntuple->Fill();
+      *fieldI32 = 1;
+      fieldKlass->s = "abc";
+      ntuple->Fill();
+   }
+
+   auto model = RNTupleModel::Create();
+   auto fieldI32 = std::make_unique<RField<std::int32_t>>("i32");
+   auto fieldKlass = std::make_unique<RField<CustomStruct>>("klass");
+   RFieldCallbackInjector::Inject(*fieldI32, [](RFieldValue &value) {
+      static std::int32_t expected = 0;
+      EXPECT_EQ(*value.Get<std::int32_t>(), expected++);
```

Comment:
Let's also modify the value and verify that after `LoadEntry()` the default entry's value reflects the modification.